### PR TITLE
Group 3: initial commit, not working, to do later

### PR DESCRIPTION
**Pull request by bdauer**
Amir and Ben.

Didn't manage to filter the junk files from c9 with .gitignore. Will look at later. Also, script is completely broken. We both work full time. Probably won't have any changes until tonight.

### CODE REVIEW

File: linked_list/.~c9_invoke_3jhEI2.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    """
+
+    def __init__(self, elem, next=None):
+        self.elem = elem
+        self.next = next
+
+    def __str__(self):
+        return str(self.elem)
+
+    def __eq__(self, other):
```
Comment (ghost) [inline linked_list/.~c9_invoke_3jhEI2.py]: I personally wasn't able to do much, but from what I understood I wonder if you have to make not just the node equal but its connection also equal.

File: linked_list/.~c9_invoke_3jhEI2.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    def __repr__(self):
+        return repr(self.elem)
+
+
+class LinkedList(AbstractLinkedList):
+    """
+    Implementation of an AbstractLinkedList inteface.
+    """
+
+    def __init__(self, elements=None):
```
Comment (ghost) [inline linked_list/.~c9_invoke_3jhEI2.py]: I was confused if we were suppose to use an empty list at all.  But like i said before I wasn't able to do much myself, in all honesty your code is clean and readable.

File: .gitignore [hunk 3]
```diff
@@ -60,3 +60,7 @@ target/
 
 #Ipython Notebook
 .ipynb_checkpoints
+
+# Cloud9 Junk
+
+.~c9
```
Comment (phillipdwright) [inline .gitignore]: I think you have to put `.~c9*` to match any file that starts with '.~c9'.

Before running `git add` on cloud9, if you run `git status`, git will print a list of all files it recognizes as being modified that are candidates for staging.  This has saved me a couple times when I've forgotten to add something to .gitignore or typed it in wrong.

**Comment by phillipdwright** [comment]
It looks like you have this built around a skeleton of an `elements` list that is stored as an attribute to each `LinkedList` instance (which you have removed).  We did that, too, at first, and it was easy to derive the rest of the magic methods directly from it.

We found that most of the other magic methods depended on being able to iterate over an object of the class.  The tough part was writing code to make the `LinkedList` iterable without a backbone list object, but once we had `__iter__()` working, the rest of the methods fell into place.  I would recommend starting with that.  [The Rafe Kettler site](http://www.rafekettler.com/magicmethods.html#sequence) we looked at in class was kind of helpful, but seeing code examples was really more helpful to me.  (I just googled to find more examples of `__iter__()` methods.)

**Comment by bdauer** [comment]
Thanks for all of your comments guys! Really helpful!

**Comment by phillipdwright** [comment]
Sure thing!  Done reviewing.

**Pull request closed without merging**